File: android/app/src/main/java/com/siansiansu/taigikeyboard/ime/dictionary/InputType.kt
// region Shared-Core Candidate
// Pure logic, Kotlin stdlib only. Eligible for cross-platform extraction.
// endregion

// 中文: 字典查詢輸入類型 — RomanWithoutTone / RomanWithTone / Hanzi 三種。
// 中文: 對應 proto InputType。由 LexiconBridge.classifyInput 實際分類。

package com.siansiansu.taigikeyboard.ime.dictionary

/**
 * Enum representing the type of input for dictionary search
 */
sealed class InputType {
    /** Search by romanization without tone marks (e.g., "goa") */
    object RomanWithoutTone : InputType()

    /** Search by romanization with tone marks (e.g., "góa") */
    object RomanWithTone : InputType()

    /** Search by Chinese characters (e.g., "我") */
    object Hanzi : InputType()
}

File: ios/Sources/TaigiKeyboard/Lexicon/Models/InputType.swift
// Classifies a dictionary-query input, which decides which trie / index column it routes through.

import Foundation

// MARK: - Shared-Core Candidate

// Pure logic, Foundation-only. Eligible for cross-platform extraction.

enum InputType {
    case romanWithoutTone // "goa" → searches poj_no_tone/tl_no_tone
    case romanWithTone // "góa" → searches poj/tl
    case hanzi // "我" → searches hanzi
}
